sklearnex/linear_model/linear.py

```diff
@@ -17,365 +17,341 @@
-            }
-
-            def __init__(
-                self,
-                fit_intercept=True,
-                copy_X=True,
-                n_jobs=None,
-                positive=False,
-            ):
-                super().__init__(
-                    fit_intercept=fit_intercept,
-                    copy_X=copy_X,
-                    n_jobs=n_jobs,
-                    positive=positive,
-                )
-
-        elif sklearn_check_version("0.24"):
-
-            def __init__(
-                self,
-                fit_intercept=True,
-                normalize="deprecated" if sklearn_check_version("1.0") else False,
-                copy_X=True,
-                n_jobs=None,
-                positive=False,
-            ):
-                super().__init__(
-                    fit_intercept=fit_intercept,
-                    normalize=normalize,
-                    copy_X=copy_X,
-                    n_jobs=n_jobs,
-                    positive=positive,
-                )
-
-        else:
-
-            def __init__(
-                self,
-                fit_intercept=True,
-                normalize=False,
-                copy_X=True,
-                n_jobs=None,
-            ):
-                super().__init__(
-                    fit_intercept=fit_intercept,
-                    normalize=normalize,
-                    copy_X=copy_X,
-                    n_jobs=n_jobs,
-                )
-
-        def fit(self, X, y, sample_weight=None):
-            """
-            Fit linear model.
-            Parameters
-            ----------
-            X : {array-like, sparse matrix} of shape (n_samples, n_features)
-                Training data.
-            y : array-like of shape (n_samples,) or (n_samples, n_targets)
-                Target values. Will be cast to X's dtype if necessary.
-            sample_weight : array-like of shape (n_samples,), default=None
-                Individual weights for each sample.
-                .. versionadded:: 0.17
-                   parameter *sample_weight* support to LinearRegression.
-            Returns
-            -------
-            self : object
-                Fitted Estimator.
-            """
-            if sklearn_check_version("1.0"):
-                self._check_feature_names(X, reset=True)
-            if sklearn_check_version("1.2"):
-                self._validate_params()
-
-            dispatch(
-                self,
-                "fit",
-                {
-                    "onedal": self.__class__._onedal_fit,
-                    "sklearn": sklearn_LinearRegression.fit,
-                },
-                X,
-                y,
-                sample_weight,
+            self._validate_params()
+
+        # It is necessary to properly update coefs for predict if we
+        # fallback to sklearn in dispatch
+        if hasattr(self, "_onedal_estimator"):
+            del self._onedal_estimator

```

Comment:
Please, add more description why exactly this manual removal is needed.

Reply:
it is necessary since i added ability to run estimator which was not trained by onedal. for example we run fit and it fall back to sklearn. it means that _onedal_estimator.model will not be updated. but if it existed from previous fit/predict call it means that predict will be run on old model. there is test in sklearn which triggers such behavior and it will fail without this change